src/vm/argdestination.h

```diff
@@ -105,6 +94,33 @@ class ArgDestination
         STATIC_CONTRACT_MODE_COOPERATIVE;
 
         _ASSERTE(IsStructPassedInRegs());
+
+#if defined (_TARGET_ARM64_)
+        // We are either copying either a float or double HFA and need to
+        // enregister each field.
+
+        int floatRegCount = m_argLocDescForStructInRegs->m_cFloatReg;
+        bool typeFloat = m_argLocDescForStructInRegs->m_isSinglePrecision;
+        void* dest = this->GetDestinationAddress();
+
+        if (typeFloat)
+        {
+            // Zero out memory before copying.
+            memset(dest, 0, floatRegCount * sizeof(double));
```

Comment:
Why do we zero the memory when we use 64 bit stores in the loop below that overwrite all the zeroed bytes anyways?